tests/testswitch.nim

```diff
@@ -714,12 +714,12 @@ suite "Switch":
 
     await allFuturesThrowing(readers)
     await switch2.stop() #Otherwise this leeks
+    await switch1.stop()
```

Comment:
The problem of moving the stop here is that the tracker checks don't make any sense after it. What this is testing for is that after stopping `switch1`, all resources are released on both, ends - meaning that channels, secure connections and chronos streams are both tore down.

If something is tripping the trackers, it means that the teardown flow broke.

Reply:
This is likely because MacOS takes some time to close the stream on the receiving end, and thus the stream isn't closed quickly enough, since adding a sleepAsync in the middle fixes the issue (https://github.com/status-im/nim-libp2p/commit/438a757c8d793b66a275f769e9e686a2de540669 , don't mind the allFuturesThrowing moving around)

Not sure how to fix the test properly, a sleepAsync is not great but does the job

Reply:
Ah, yeah this might very well be the case, does it happen on linux & windows as well?

Reply:
No, at least not on my linux laptop

Reply:
Great feedback @dryajov and @Menduist. I'll remove this last commit from the PR; continuing the conversation in #592.